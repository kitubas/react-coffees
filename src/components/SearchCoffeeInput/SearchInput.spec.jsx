import { render, screen } from '@testing-library/react';
import { SearchCoffeeInput } from '.';
import userEvent from '@testing-library/user-event';

describe('<SearchCoffeeInput />', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<SearchCoffeeInput handleSearchButtonChange={fn} searchValue={'testando'} />);

    const input = screen.getByPlaceholderText(/procure o café pelo nome/i);
    expect(input.value).toBe('testando');
  });

  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<SearchCoffeeInput handleSearchButtonChange={fn} searchValue="um valor" />);

    const input = screen.getByPlaceholderText(/procure o café pelo nome/i);
    const value = 'outro valor';

    userEvent.type(input, value);

    expect(input.value).toBe('um valor');
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should match snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<SearchCoffeeInput handleSearchButtonChange={fn} searchValue="um valor" />);
    expect(container).toMatchSnapshot();
  });
});
