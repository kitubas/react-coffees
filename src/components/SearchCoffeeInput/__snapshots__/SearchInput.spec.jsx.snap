// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchCoffeeInput /> should match snapshot 1`] = `
<div>
  <input
    class="search-coffee-input"
    placeholder="procure o cafÃ© pelo nome"
    type="search"
    value="um valor"
  />
</div>
`;
